src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+
+namespace opossum {
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
```

Comment:
We place (most) classes in their own file. Please check for other classes as well.

Reply:
This is marked as outdated but has not been addressed yet.